File: frontend/src/constants/enums.ts
export const TaskStatus = {
  TO_DO: "to_do",
  IN_PROGRESS: "in_progress",
  DONE: "done",
  BLOCKED: "blocked",
} as const;

export type TaskStatus = (typeof TaskStatus)[keyof typeof TaskStatus];

export const TaskPriority = {
  LOW: "low",
  MEDIUM: "medium",
  HIGH: "high",
  CRITICAL: "critical",
} as const;

export type TaskPriority = (typeof TaskPriority)[keyof typeof TaskPriority];

export const ProjectStatus = {
  ACTIVE: "active",
  ARCHIVED: "archived"
} as const;

export type ProjectStatus = (typeof ProjectStatus)[keyof typeof ProjectStatus];

File: backend/src/models/enums.py
import enum


class UserRole(str, enum.Enum):
    admin = "admin"
    manager = "manager"
    member = "member"


class ProjectStatus(str, enum.Enum):
    active = "active"
    archived = "archived"


class TaskStatus(str, enum.Enum):
    to_do = "to_do"
    in_progress = "in_progress"
    done = "done"
    blocked = "blocked"


class TaskPriority(str, enum.Enum):
    low = "low"
    medium = "medium"
    high = "high"
    critical = "critical"